packages/office-ui-fabric-react/src/components/DocumentCard/DocumentCardTitle.base.tsx

```diff
@@ -95,72 +100,84 @@ export class DocumentCardTitleBase extends BaseComponent<IDocumentCardTitleProps
     return documentCardTitle;
   }
 
-  private _startTruncation = (props: IDocumentCardTitleProps): void => {
-    const originalTitle = props.title;
-    this._isTruncated = false;
-    const miniLength = props.showAsSecondaryTitle ? TRUNCATION_MINI_LENGTH_SECONDARY : TRUNCATION_MINIMUM_LENGTH;
-    const maxLength = props.showAsSecondaryTitle ? TRUNCATION_MAX_LENGTH_SECONDARY : TRUNCATION_MAXIMUM_LENGTH;
-
-    // If the title is really short, there's no need to truncate it
-    if (originalTitle && originalTitle.length >= miniLength) {
-      // Break the text into two pieces for assembly later
-      if (originalTitle.length > maxLength) {
-        // The text is really long, so we can take a chunk out of the middle so the two pieces combine for the maximum length
-        this._isTruncated = true;
-        this.setState({
-          truncatedTitleFirstPiece: originalTitle.slice(0, maxLength / 2 + TRUNCATION_FIRST_PIECE_LONGER_BY),
-          truncatedTitleSecondPiece: originalTitle.slice(originalTitle.length - (maxLength / 2 - TRUNCATION_FIRST_PIECE_LONGER_BY))
-        });
-      } else {
-        // The text is not so long, so we'll just break it into two pieces
-        this.setState({
-          truncatedTitleFirstPiece: originalTitle.slice(0, Math.ceil(originalTitle.length / 2) + TRUNCATION_FIRST_PIECE_LONGER_BY),
-          truncatedTitleSecondPiece: originalTitle.slice(
-            originalTitle.length - Math.floor(originalTitle.length / 2) + TRUNCATION_FIRST_PIECE_LONGER_BY
-          )
-        });
-      }
+  // Truncate logic here way can't handle the case that chars with different widths are mixed very well, let _shrinkTitle take care of that.
+  private _truncateTitle = (): void => {
+    if (!this.state.needMeasurement) {
+      return;
     }
 
-    // Save the width we just started truncation at, so that later we will only update truncation if necessary
-    if (this._titleElement.current) {
-      this._truncatedTitleAtWidth = this._titleElement.current.clientWidth;
+    this._async.requestAnimationFrame(this._truncateWhenInAnimation);
+  };
+
+  private _truncateWhenInAnimation: () => void = () => {
+    const originalTitle = this.props.title;
+    const element: HTMLDivElement | null = this._measureTitleElement.current;
+
+    if (element) {
+      const style: CSSStyleDeclaration = getComputedStyle(element);
+      if (style.width && style.lineHeight && style.height) {
+        const { clientWidth, scrollWidth } = element;
+        const lines: number = Math.floor(
+          (parseInt(style.height, 10) + TRUNCATION_VERTICAL_OVERFLOW_THRESHOLD) / parseInt(style.lineHeight, 10)
+        );
+        // Used to calculate the number of chars that should be truncated.
+        const overFlowRate: number = scrollWidth / (parseInt(style.width, 10) * lines);
+
+        if (overFlowRate > 1) {
+          const truncatedLength: number = originalTitle.length / overFlowRate - 3 /** Saved for separator */;
```

Comment:
Why -3? Does this particular number mean something?

Reply:
As explained in the comments, it was saved for `...` which length is 3

Reply:
I added comments to explain the whole logic